sources/packages/backend/apps/queue-consumers/src/processors/virus-scan/_tests_/virus-scan.processor.e2e-spec.ts

```diff
@@ -40,12 +46,147 @@ describe(describeProcessorRootTest(QueueNames.FileVirusScanProcessor), () => {
     jest.resetAllMocks();
   });
 
+  it("Should throw an error when the student file is not found during virus scanning.", async () => {
+    // Arrange
+    const studentFile = createFakeStudentFileUpload();
+    studentFile.virusScanStatus = VirusScanStatus.InProgress;
+    await db.studentFile.save(studentFile);
+    clamAVServiceMock.scanFile = jest.fn(() =>
+      Promise.resolve({
+        isInfected: null,
+        errorCode: FILE_NOT_FOUND,
+      }),
+    ); // Queued job.
+    const mockedJob = mockBullJob<VirusScanQueueInDTO>({
+      uniqueFileName: "file.random",
+      fileName: studentFile.fileName,
+    });
+
+    // Act
+    const errorMessage =
+      "File file.random is not found or has already been scanned for viruses. Scanning the file for viruses is aborted.";
+    await expect(
+      processor.performVirusScan(mockedJob.job),
+    ).rejects.toStrictEqual(new Error(errorMessage));
+    expect(
+      mockedJob.containLogMessages([
+        "Log details",
+        "Starting virus scan.",
+        "ClamAV server availability: uncertain.",
```

Comment:
Is there a need to report the "server availability" every time?
If not available the error details will mention it, right?